song-server/src/main/java/bio/overture/song/server/utils/Scopes.java

```diff
@@ -34,7 +43,29 @@ public static long extractExpiry(Map<String, ?> map) {
     return 0L;
   }
 
-  private static Set<String> getScopes(OAuth2Authentication o2auth) {
-    return o2auth.getOAuth2Request().getScope();
+  private static Set<String> getJwtScopes(JwtAuthenticationToken jwt) {
+    Set<String> output = new HashSet();
+    try {
+      val context = jwt.getToken().getClaim("context");
```

Comment:
can context be null here ?

Reply:
instanceof returns false when context is null. should work as expected.